lib/spack/docs/packaging_guide.rst

```diff
@@ -143,139 +124,133 @@ file in your favorite ``$EDITOR``:
    from spack import *
 
 
-   class Cmake(Package):
+   class Gmp(AutotoolsPackage):
        """FIXME: Put a proper description of your package here."""
 
        # FIXME: Add a proper url for your package's homepage here.
        homepage = "http://www.example.com"
-       url      = "http://www.cmake.org/files/v2.8/cmake-2.8.12.1.tar.gz"
+       url      = "https://gmplib.org/download/gmp/gmp-6.1.2.tar.bz2"
 
-       version('2.8.12.1', '9d38cd4e2c94c3cea97d0e2924814acc')
-       version('2.8.12',   '105bc6d21cc2e9b6aff901e43c53afea')
-       version('2.8.11.2', '6f5d7b8e7534a5d9e1a7664ba63cf882')
+       version('6.1.2', '8ddbb26dc3bd4e2302984debba1406a5')
+       version('6.1.1', '4c175f86e11eb32d8bf9872ca3a8e11d')
+       version('6.1.0', '86ee6e54ebfc4a90b643a65e402c4048')
 
-       # FIXME: Add dependencies if this package requires them.
-       # depends_on("foo")
+       # FIXME: Add dependencies if required.
```

Comment:
I'm kind of tempted to say we should just not bother at that point.  I think it'll be confusing to have two autotools packages, and the existing one already has an `autoreconf` phase.  Also, I think it's complicated to do this generically.  Some packages use `automake` but not `autoconf`, some do it the other way around, and some may or may not use `libtool`.  The particular dependencies you want are going to depend on that stuff, so I don't think it's unreasonable to require the packager to straighten this stuff out for packages that don't ship with a proper configure script.

I think we should be mindful of making the common case easy and the special cases possible.  This accomplishes that very well, and we go a step further by automating a number of important build systems.  I think helping people beyond that is outside the scope of what we should have to maintain in Spack.